plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialLevel.java

```diff
@@ -2,53 +2,69 @@
 
 import com.denizenscript.denizen.objects.MaterialTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.ElementTag;
-import com.denizenscript.denizencore.objects.properties.Property;
 import com.denizenscript.denizencore.objects.properties.PropertyParser;
 import org.bukkit.block.data.BlockData;
+import org.bukkit.block.data.Brushable;
 import org.bukkit.block.data.Levelled;
 import org.bukkit.block.data.type.Beehive;
 import org.bukkit.block.data.type.Cake;
 import org.bukkit.block.data.type.Farmland;
 import org.bukkit.block.data.type.Snow;
 
-public class MaterialLevel implements Property {
-
-    public static boolean describes(ObjectTag material) {
-        if (!(material instanceof MaterialTag)) {
-            return false;
-        }
-        MaterialTag mat = (MaterialTag) material;
-        if (!mat.hasModernData()) {
-            return false;
-        }
-        BlockData data = mat.getModernData();
-        return data instanceof Levelled
-                || data instanceof Cake
-                || data instanceof Snow
-                || data instanceof Farmland
-                || data instanceof Beehive;
+public class MaterialLevel extends MaterialProperty<ElementTag> {
+
+    // <--[property]
+    // @object MaterialTag
+    // @name level
+    // @input ElementTag(Number)
+    // @description
+    // Controls the current level for a Levelled material, cake, beehives, snow, farmland, or brushable blocks.
+    // "Levelled" materials include: water, lava, cauldrons, composters, light blocks, brushable blocks, and any other future Levelled implementing types.
+    // For light blocks, this is the brightness of the light.
+    // For water/lava this is the height of the liquid block.
+    // For cauldrons, this is the amount of liquid contained.
+    // For cake, this is the number of bites left.
+    // For beehives/bee nests, this is the amount of honey contained.
+    // For snow, this is the number of partial layers, or the height, of a snow block.
+    // For farmland, this is the moisture level.
+    // For composters, this is the amount of compost.
+    // For brushable blocks, this is the level of dusting.
+    // See also <@link tag MaterialTag.maximum_level> and <@link tag MaterialTag.minimum_level>.
+    // -->
+
+    public static boolean describes(MaterialTag material) {
+        BlockData data = material.getModernData();
+        return data instanceof Levelled 
+                || data instanceof Cake 
+                || data instanceof Snow 
+                || data instanceof Farmland 
+                || data instanceof Beehive 
+                || data instanceof Brushable;

```

Comment:
As these blocks are new to 1.20, references to Brushable in pre-existing properties needs to be handled with care - search for examples of `NMSVersion` references in object property classes to compare with